core/src/env/srml/types.rs

```diff
@@ -44,3 +45,17 @@ impl<'a> From<&'a [u8]> for Address {
 
 /// The default SRML balance type.
 pub type Balance = u64;
+
+/// The default SRML hash type.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Encode, Decode)]
+pub struct Hash([u8; 32]);
+
+impl<'a> From<&'a [u8]> for Hash {
```

Comment:
With the approach of `TryFrom` we can do slightly better here by implementing a `impl<'a> From<&'a [u8; 32]> for Hash` instead. With `TryFrom` we can very simply convert from `&[u8]` into `&[u8;32]`.

Reply:
Can we do this in another PR for both `Hash` and `Address`?

Reply:
sure